src/Runner.Worker/ActionCommandManager.cs

```diff
@@ -223,6 +223,24 @@ public void ProcessCommand(IExecutionContext context, string line, ActionCommand
                 throw new Exception("Required field 'name' is missing in ##[set-env] command.");
             }
 
+
+            foreach (var blocked in _setEnvBlockList)
+            {
+                if (string.Equals(blocked, envName, StringComparison.OrdinalIgnoreCase))
+                {
+                    // Log Telemetry and let user know they shouldn't do this
+                    var issue = new Issue()
+                    {
+                        Type = IssueType.Error,
+                        Message = "Can't update NODE_OPTIONS environment variable using ::set-env:: command."
```

Comment:
can we make this message part of a hash or something since the actual list of env's and blocks could be generic? otherwise we should just hard code the variable too

Reply:
fixed.